modules/cnv-about-hostpath-provisioner.adoc

```diff
@@ -0,0 +1,18 @@
+// Module included in the following assemblies:
+//
+// * cnv/cnv_users_guide/cnv-configuring-local-storage-for-vms.adoc
+
+[id="cnv-about-hostpath-provisioner_{context}"]
+= About the hostpath provisioner
+
+The hostpath provisioner is a local storage provisioner designed for
+{CNVProductName}. If you want to configure local storage for
+virtual machines, you must enable the hostpath provisioner first.
+
+The Operator for the hostpath provisioner is installed by default with the
+{CNVProductName} Operator. The hostpath provisioner Operator deploys the
+provisioner itself, and it creates the required `StorageClass` object.
```

Comment:
It does not create the storage class object either, since there are 2 possible options for the binding mode, we decided to not manage the storage class with the operator. So the user will have to create the storage class as well. https://github.com/kubevirt/hostpath-provisioner-operator#storage-class has some more information on this.

Reply:
@awels Oh, I thought that users running CNV 2.2 would have to be running Kubernetes >= 1.12 since they would not be running OCP 3.x. In November you had said that the hostpath provisioner operator creates the storage class for you; can you confirm that I should add info about the two different options?

Thanks!

Reply:
Well technically there is nothing stopping people from running cnv 2.2 on OS 3.11, so that is why we had to give them the option. We don't recommend it, but its not impossible.

Reply:
So in November this was the case, but after some thought, we decided that we should also support 3.11 in case people wanted to run the hostpath provisioner on OS 3.11. Also other provisioners usually don't have the operator manage the storage class. And last but not least, we wanted to give the user the option to name the storage class differently from the default.

So I would add a section similar to the CR, where you have an example yaml and point out the things they can customize (the volumeBindingMode and name).

Reply:
@awels Thanks for explaining the reasoning behind that decision. :) I will include info about setting the storage class binding mode to `WaitForFirstConsumer`, but because we really only recommend CNV 2.2 on OCP 4+ (if not 4.3), I am going to omit the info about the `Immediate` binding mode. I also ran this by @djfjeff and he agrees that we should only include info relevant to OCP 4+.

Reply:
Agreed, I just wanted to make you aware of the reasoning and options so you could write the best possible documentation.

Reply:
Hi @awels and @alitke - I have made some changes that I would like you to review (mostly adding info about creating the storage class). Here's the latest preview build: http://file.bos.redhat.com/pousley/121719/hostpath-cdi/cnv/cnv_users_guide/cnv-configuring-local-storage-for-vms.html

And here's the commit with most of the changes: https://github.com/openshift/openshift-docs/pull/18445/commits/b0c76c831e37b7b4bb98edb019a5c5a0b3391dfe

Please let me know if you think the info is in the right order, in particular, and if there are any inaccuracies. There are some references to kubevirt (API versions, etc.) so please let me know if any of those need to be replaced with downstream versions (and what the replacements should be).

Thanks so much!